### DISCUSSION TITLE
Any way to access archive through USB?

### DISCUSSION THREAD

**Discussion by cpxazn** [marcone/teslausb#895] [General]
Is there any way to mount the archive to a PC through USB? Currently I can only access the rolling 1 hour RecentClips footage through USB, but the GUI has several days worth of footage. I am able to access through Samba, but transfer rates are 3MB/s with a pi zero. I need to transfer 3 days worth of footage so I can look through them quickly, and this transfer is estimated for over 24 hours.

**Comment by marcone** [comment]
What does "look through them quickly" mean exactly? Any reason you can't use the web interface?

In any case, accessing those clips through USB is not currently possible, however if you're tech savvy you can try https://github.com/viveris/uMTP-Responder/, which is what I'm currently trying to integrate into TeslaUSB.

**Reply by cpxazn** [reply]
> What does "look through them quickly" mean exactly? Any reason you can't use the web interface?
> 
> In any case, accessing those clips through USB is not currently possible, however if you're tech savvy you can try https://github.com/viveris/uMTP-Responder/, which is what I'm currently trying to integrate into TeslaUSB.

I have to look through 72 hours worth of footage to find a cause of damage, so I need to be able to be spam the +30s button without delay. The footage is inside a car shop and sentryclips itself has no events during that time even though there were people walking all around the car.

**Reply by marcone** [reply]
So I assume the damage is visible in the recording then? (because otherwise skipping through the recordings in 30-second steps wouldn't really help)
In that case shouldn't you be able to just do a sort of binary search using the seek bar to quickly find the moment that the damage happened?

In any case, what you're asking for is not currently possible out of the box. You can try the MTP code I linked, or disable TeslaUSB's drive emulation and instead have it emulate a network adapter so you can hopefully transfer files faster that way. If you have access to a linux machine, you can also mount the sd card on that, and then mount the individual snapshots in the /backingfiles/snapshots/ folder.
Note that even if you copied all the recordings off of the Pi, you would then be dealing with individual 1-minute segments, so you'd have to find some way to watch through those.

**Reply by cpxazn** [reply]
The damage isn't visible, but I'm looking if something like a rolling chair hit the car or someone examining that particular area after accidentally damaging it. I'll evaluate which are the best options. Thanks for the help.

**Reply by tete1030** [reply]
Hi, @marcone . Does the MTP method work for reading the filesystem while Tesla is writing to it? I'm very interested in this. Currently I'm just doing a per-minute snapshot, and it works well except for occasional fs damage within the snapshot (file being zero-sized). So I'm looking for solutions of simultaneous accessibility.

**Reply by marcone** [reply]
No, MTP won't work for that, because the Pi doesn't access the drive image while it is connected to the car.
